opentelemetry-sdk/src/opentelemetry/sdk/trace/__init__.py

```diff
@@ -321,18 +321,29 @@ def get_current_span(self):
         """See `opentelemetry.trace.Tracer.get_current_span`."""
         return self._current_span_slot.get()
 
-    @contextmanager
     def start_span(
         self,
         name: str,
         parent: trace_api.ParentSpan = trace_api.Tracer.CURRENT_SPAN,
         kind: trace_api.SpanKind = trace_api.SpanKind.INTERNAL,
-    ) -> typing.Iterator["Span"]:
+    ) -> "Span":
         """See `opentelemetry.trace.Tracer.start_span`."""
 
         span = self.create_span(name, parent, kind)
         span.start()
-        with self.use_span(span, end_on_exit=True):
+        return span
+
+    @contextmanager
+    def start_as_current_span(
```

Comment:
I've always been struggling with naming.
Given this technically is `start_span` then `use_span`, maybe we should just call it `start_and_use_span`❓

Reply:
I'd actually argue that "start_span" should remain as is, and we should maybe provide separate terminology for the two components of span creation and use.

I don't imagine most consumers will really need to separate span creation and use, and may find it as needlessly verbose for "start_and_use_span".

Reply:
This is also about consistency across languages.